hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/ShowInvalidParquetProcedure.scala

```diff
@@ -52,11 +54,14 @@ class ShowInvalidParquetProcedure extends BaseProcedure with ProcedureBuilder {
     val srcPath = getArgValueOrDefault(args, PARAMETERS(0)).get.asInstanceOf[String]
     val limit = getArgValueOrDefault(args, PARAMETERS(1))
     val needDelete = getArgValueOrDefault(args, PARAMETERS(2)).get.asInstanceOf[Boolean]
+    val partitions = getArgValueOrDefault(args, PARAMETERS(3)).map(_.toString).getOrElse("")
     val storageConf = HadoopFSUtils.getStorageConfWithCopy(jsc.hadoopConfiguration())
     val storage = new HoodieHadoopStorage(srcPath, storageConf)
-    val partitionPaths: java.util.List[String] = FSUtils.getAllPartitionPaths(
-      new HoodieSparkEngineContext(jsc), storage, srcPath, false)
-    val javaRdd: JavaRDD[String] = jsc.parallelize(partitionPaths, partitionPaths.size())
+    val metadataConfig = HoodieMetadataConfig.newBuilder.enable(false).build
+    val metadata = HoodieTableMetadata.create(new HoodieSparkEngineContext(jsc), storage, metadataConfig, srcPath)
+    val partitionPaths: java.util.List[String] = metadata.getPartitionPathWithPathPrefixes(partitions.split(",").toList.asJava)
```

Comment:
We should use original `FSUtils.getAllPartitionPaths(
      new HoodieSparkEngineContext(jsc), storage, srcPath, false)` when users didn't set `partitions` config

Reply:
`getPartitionPathWithPathPrefixes(partitions)` will return all partitionPaths when `partitions` are empty.